en/components/grid.md

```diff
@@ -234,141 +310,289 @@ export class MyComponent implements OnInit {
+The grid supports both horizontal and vertical virtualization. In order to activate it make sure to set either the `height` or `width` property of the grid.
 
-Corresponding public methods are exposed for developers to perform CRUD operations:
+### CRUD operations
+
+The `IgxGridComponent` provides a straigtforward API for basic CRUD operations.
+
+#### Adding a new record
+
+The grid component exposes the `addRow` method which will add the provided data to the data source itself.
 
 ```typescript
-public addRow() {
-    const record = {ProductID: this.grid1.data[this.grid1.data.length - 1].ProductID + 1, ProductName: 'Camembert Pierrot'};
-    this.grid1.addRow(record);
-}
+// Adding a new record
+// Assuming we have a `getNewRecord` method returning the new row data.
+const record = this.getNewRecord();
+this.grid.addRow(record);
+```
 
-public updateRecord(event) {
-    this.grid1.updateCell(this.selectedCell.rowIndex, this.selectedCell.columnField, event);
-    this.grid1.getCell(this.selectedCell.rowIndex, this.selectedCell.columnField);
-}
+#### Updating data in the grid
 
-public deleteRow(event) {
-    this.selectedRow = Object.assign({}, this.grid1.getRow(this.selectedCell.rowIndex));
-    this.grid1.deleteRow(this.selectedCell.rowIndex);
-    this.selectedCell = {};
-    this.snax.message = `Row with ID ${this.selectedRow.record.ID} was deleted`;
-    this.snax.show();
-}
+Updating data in the grid is achieved through `updateRow` and `updateCell` methods. You can also directly update a cell value through its `update` method.
+
+```typescript
+// Updating the whole row
+this.grid.updateRow(newData, this.selectedCell.rowIndex);
+
+// Just a particualr cell through the Grid API
+this.grid.updateCell(newData, this.selectedCell.rowIndex, this.selectedCell.column.field);
+
+// Directly using the cell `update` method
+this.selectedCell.update(newData);
 ```
 
-These can be wired to user interactions, not necessarily related to the **igx-grid**; for example, a button click:
+#### Deleting data from the grid
 
+```typescript
+this.grid.deleteRow(this.selectedCell.rowIndex);
+```
+These can be wired to user interactions, not necessarily related to the **igx-grid**; for example, a button click:
 ```html
 <button igxButton igxRipple (click)="deleteRow($event)">Delete Row</button>
 ```
 
-<div class="divider--half"></div>
-
 ### Paging
 
-**Paging** is initialized on the root **igx-grid** component, and is configurable via the `paging` and `perPage` inputs. Paging is a Boolean property that controls whether the feature is enabled and the perPage property controls the visible records per page. Let’s update our grid to enable paging:
+**Paging** is initialized on the root **igx-grid** component, and is configurable via the `paging` and `perPage` inputs. Paging is a Boolean property that controls whether the feature is enabled and the `perPage` property controls the visible records per page. Let’s update our grid to enable paging:
```

Comment:
Note that the `perPage` input has to be a number larger than the virtualization chunk size.